Source/WebCore/platform/mediastream/cocoa/AudioMediaStreamTrackRendererUnit.h

```diff
@@ -50,13 +50,19 @@ class AudioSampleDataSource;
 class AudioSampleBufferList;
 class CAAudioStreamDescription;
 
-class AudioMediaStreamTrackRendererUnit : public BaseAudioMediaStreamTrackRendererUnit, public CanMakeWeakPtr<AudioMediaStreamTrackRendererUnit, WeakPtrFactoryInitialization::Eager>, AudioMediaStreamTrackRendererInternalUnit::Client {
+class AudioMediaStreamTrackRendererUnit : public BaseAudioMediaStreamTrackRendererUnit, public CanMakeWeakPtr<AudioMediaStreamTrackRendererUnit, WeakPtrFactoryInitialization::Eager>, public AudioMediaStreamTrackRendererInternalUnit::Client {
 public:
     WEBCORE_EXPORT static AudioMediaStreamTrackRendererUnit& singleton();
 
     AudioMediaStreamTrackRendererUnit();
     ~AudioMediaStreamTrackRendererUnit();
 
+    // Since AudioMediaStreamTrackRendererUnit is a singleton, it's ref/deref do nothing.
+    void ref() const final { }
+    void deref() const final { }
+
+    USING_CAN_MAKE_WEAKPTR(AudioMediaStreamTrackRendererInternalUnit::Client);
```

Comment:
No, you need to remove the `CanMakeWeakPtr<AudioMediaStreamTrackRendererUnit, WeakPtrFactoryInitialization::Eager>` base class. Also, you'll likely need to use `WeakPtrFactoryInitialization::Eager` on the client as well, otherwise you may break things.

Reply:
Ah ok that makes sense. Fixed!